### PR TITLE
show message for bad token error in map plugin

### LINKED ISSUE #2363: [BUG] Show error message to user when invalid mapbox token is provided
When an invalid mapbox token is provided, the map fails to load but no error message is shown to the user. A better behavior would be to surface the underlying issue to the user:

```py
import fiftyone as fo
import fiftyone.zoo as foz

dataset = foz.load_zoo_dataset("quickstart-geo")

dataset.app_config["plugins"] = {"map": {"mapboxAccessToken": "bad-token"}}
dataset.save()

session = fo.launch_app(dataset)
```

Screen Shot 2022-11-28 at 4 57 58 PM

### PR DESCRIPTION
**Pull request by lanzhenw**
## What changes are proposed in this pull request?
fixes #2363 

Handle the map token error when rendering the map
Screenshot 2022-12-02 at 11 37 56 AM

## How is this patch tested? If it is not, please explain why.

Local

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: app/packages/map/src/Map.tsx [hunk 1]
```diff
@@ -170,6 +171,8 @@ const Plot: React.FC<{}> = () => {
     <div className={container} ref={ref}>
       {loading && !length ? (
         <foc.Loading style={{ opacity: 0.5 }}>Pixelating...</foc.Loading>
+      ) : mapError ? (
+        <foc.Loading> Bad token </foc.Loading>
```
Comment (benjaminpkane) [inline app/packages/map/src/Map.tsx:175]: Maybe 
```
        <foc.Loading>An expected map error occurred. The Mapbox token may be invalid</foc.Loading>
```
cc @brimoor

File: app/packages/map/src/Map.tsx [hunk 2]
```diff
@@ -190,7 +193,10 @@ const Plot: React.FC<{}> = () => {
               if (draw.getMode() !== "draw_polygon") {
                 draw.changeMode("draw_polygon");
               }
-            } catch {}
+            } catch (error) {
+              console.error(error);
+              setMapError(true);
```
Comment (benjaminpkane) [inline app/packages/map/src/Map.tsx:198]: Maybe the below will be easier for people to debug?
```suggestion
              setMapError(true);
              throw error
```

Comment (brimoor) [reply app/packages/map/src/Map.tsx:209]: 👍

File: app/packages/map/src/Map.tsx (hunk 1 above)
Comment (brimoor) [reply app/packages/map/src/Map.tsx:186]: Yeah I agree; I committed this change

**Review by brimoor — APPROVED** [review]
LGTM. Note that I added Ben's suggestions:

Screen Shot 2022-12-05 at 10 02 43 AM

**Pull request merged**